tls/extensions/s2n_server_certificate_request_extensions.c

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "error/s2n_errno.h"
+#include "s2n_server_signature_algorithms.h"
+#include "utils/s2n_safety.h"
+#include "stuffer/s2n_stuffer.h"
+
+#include "tls/s2n_tls.h"
+#include "tls/s2n_tls13.h"
+
+int s2n_server_certificate_request_extensions_parse(struct s2n_connection *conn, struct s2n_blob *extensions)
+{
+    bool processed_sig_algs = 0;
+    struct s2n_stuffer ext_stuffer = {0};
+    GUARD(s2n_stuffer_init(&ext_stuffer, extensions));
+    GUARD(s2n_stuffer_write(&ext_stuffer, extensions));
+
+    while (s2n_stuffer_data_available(&ext_stuffer)) {
+        struct s2n_blob ext = {0};
+        uint16_t extension_type, extension_size;
+        struct s2n_stuffer current_extension = {0};
+
+        S2N_ERROR_IF(s2n_stuffer_data_available(&ext_stuffer) < 4, S2N_ERR_BAD_MESSAGE);
+        GUARD(s2n_stuffer_read_uint16(&ext_stuffer, &extension_type));
+        GUARD(s2n_stuffer_read_uint16(&ext_stuffer, &extension_size));
+
+        ext.size = extension_size;
+        ext.data = s2n_stuffer_raw_read(&ext_stuffer, ext.size);
```

Comment:
Do you need to check that the stuffer has at least ext.size data left?

Reply:
Added a check